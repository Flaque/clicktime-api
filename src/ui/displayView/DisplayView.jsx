import React from 'react';

class DisplayView extends React.Component {

  constructor(props) {
    super(props)
    console.log(this.props.selectedTask)
  }

  renderJobs() {
    if (!this.props.selectedTask.jobs) {
      return ( <p> Nothing! </p>)
    }

    return this.props.selectedTask.jobs.map( (job) =>
      <div className="row content" key={job.JobID} >
        <div className="four columns">
          {job.DisplayName}
        </div>
        <div className="four columns">
          BMW
        </div>
        <div className="four columns">
          ‚ùå
        </div>
      </div>
    )
  }

  render() {
    return (
      <div className="displayView">

        {/* Headers */}
        <div className="row">
          <div className="four columns">
            <h6 className="upper light"> Job </h6>
          </div>
          <div className="four columns">
            <h6 className="upper light"> Client </h6>
          </div>
          <div className="four columns">
            <h6 className="upper light"> Recent? </h6>
          </div>
        </div>

        {this.renderJobs()}

      </div>
    )
  }
}

export default DisplayView;
