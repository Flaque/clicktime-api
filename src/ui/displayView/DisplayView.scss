.display-empty {
  text-align: center;
  i.fa {
    color: #ddd;
    font-size: 2rem;
  }
}
