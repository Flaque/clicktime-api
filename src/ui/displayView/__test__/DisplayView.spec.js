import React from 'react';
import DisplayView from '../DisplayView.jsx';
import renderer from 'react-test-renderer';

const mockTask = {
  DisplayName: "task",
  jobs: [{
    DisplayName: "job",
    JobID: '1',
    client: {
      DisplayName: "client"
    }
  }]
}

test('Test that DisplayView is still the same', () => {
  const tree = renderer.create(
    <DisplayView selectedTask={mockTask} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
