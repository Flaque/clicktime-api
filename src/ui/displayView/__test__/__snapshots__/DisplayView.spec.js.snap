exports[`test Test that DisplayView is still the same 1`] = `
<div
  className="displayView">
  <div>
    <div
      className="row">
      <div
        className="six columns">
        <h6
          className="upper light">
           Job 
        </h6>
      </div>
      <div
        className="six columns">
        <h6
          className="upper light">
           Client 
        </h6>
      </div>
    </div>
    <div
      className="row content">
      <div
        className="six columns">
        job
      </div>
      <div
        className="six columns">
        client
      </div>
    </div>
  </div>
</div>
`;
